sycl/source/stream.cpp

```diff
@@ -13,14 +13,20 @@
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 
+// Maximum possible size of a flush buffer statement in bytes
+static constexpr size_t MAX_STATEMENT_SIZE =
+    (1 << (CHAR_BIT * detail::FLUSH_BUF_OFFSET_SIZE)) - 1;
+
 stream::stream(size_t BufferSize, size_t MaxStatementSize, handler &CGH)
     : impl(std::make_shared<detail::stream_impl>(BufferSize, MaxStatementSize,
                                                  CGH)),
       GlobalBuf(impl->accessGlobalBuf(CGH)),
       GlobalOffset(impl->accessGlobalOffset(CGH)),
       // Allocate the flush buffer, which contains space for each work item
       GlobalFlushBuf(impl->accessGlobalFlushBuf(CGH)),
-      FlushBufferSize(MaxStatementSize) {
+      FlushBufferSize(MaxStatementSize + detail::FLUSH_BUF_OFFSET_SIZE) {
+  assert((MaxStatementSize <= MAX_STATEMENT_SIZE) &&
+         "Maximum statement size too large.");
```

Comment:
Assert will be triggered only if assertions are enabled. Should it be an exception? We're violating the spec by limiting of statement size though it's nice for user to see the limitation exception instead of possible UB with silently ignoring the limit.
BTW, what's the expected result if the limit is exceeded, i.e. when assertions are disabled?

Reply:
Changed on exception throwing.